src/System.Diagnostics.StackTrace/src/System/Diagnostics/StackTrace.netcore50aot.cs

```diff
@@ -0,0 +1,49 @@
+﻿using System;
+using Internal.Diagnostics;
+
+namespace System.Diagnostics
+{
+    public sealed class StackTrace
+    {
+        private readonly IntPtr[] _ips;
+        private readonly bool _fNeedFileInfo;
+
+        public StackTrace(Exception exception, bool needFileInfo)
+        {
+            if (exception == null)
+            {
+                throw new ArgumentNullException("exception");
+            }
+            _ips = ExceptionExtensions.GetStackIPs(exception);
+            _fNeedFileInfo = needFileInfo;
+        }
+
+        public StackFrame[] GetFrames()
+        {
+            // There are special IPs mixed in to the list that aren't intended to be exposed as StackFrames
+            // Deliberately doing this without Linq to keep the dependency set down.
+            StackFrame[] tempFrames = new StackFrame[_ips.Length];
+            int countNonSpecialIPs = 0;
+            for (int i = 0; i < _ips.Length; i++)
+            {
+                if (_ips[i] != StackTraceHelper.SpecialIP.EdiSeparator)
+                {
+                    tempFrames[countNonSpecialIPs++] = new StackFrame(_ips[i], _fNeedFileInfo);
+                }
+            }
+            //for compat with CoreCLR, we always return null instead of 0 length array
+            if (countNonSpecialIPs == 0)
+            {
+                return null;
+            }
+            StackFrame[] frames = new StackFrame[countNonSpecialIPs];
+            Array.Copy(tempFrames, frames, countNonSpecialIPs);
```

Comment:
Or, use `Array.Resize`...

Something like the following (including other feedback):

```c#
public StackFrame[] GetFrames()
{
    // There are special IPs mixed in to the list that aren't intended to be exposed as StackFrames
    // Deliberately doing this without Linq to keep the dependency set down.
    IntPtr[] ips = _ips;
    StackFrame[] frames = new StackFrame[ips.Length];
    int countNonSpecialIPs = 0;
    for (int i = 0; i < ips.Length; i++)
    {
        if (ips[i] != StackTraceHelper.SpecialIP.EdiSeparator)
        {
            frames[countNonSpecialIPs++] = new StackFrame(ips[i], _needFileInfo);
        }
    }
    // For compat with CoreCLR, we always return null instead of 0 length array
    if (countNonSpecialIPs == 0)
    {
        return null;
    }
    Array.Resize(ref frames, countNonSpecialIPs);
    return frames;
}
```